vars/stageDeployToOpenshift.groovy

```diff
@@ -5,13 +5,34 @@ def call(def context) {
       return
     }
 
-    openshiftTag(
-      srcStream: context.componentId,
-      srcTag: context.tagversion,
-      destStream: context.componentId,
-      destTag: "latest",
-      namespace: context.targetProject
-    )
+    if (fileExists('openshift')) {
+      // Output version to make it easier to diagnose issues.
+      sh "tailor version"
+      // Ensure that no automatic triggers exist as we rollout manually later on.
+      sh "oc --namespace ${context.targetProject} set triggers dc/${context.componentId} --manual"
+      // Detect any <project>.env, Tailorfile.<project> or Tailorfile files.
+      def paramFile = ''
+      if (fileExists("openshift/${context.targetProject}.env")) {
+        paramFile = "--param-file ${context.targetProject}.env"
+      }
+      def tailorFlagsAndArgs = "--non-interactive --namespace ${context.targetProject} --selector app=${context.projectId}-${context.componentId} update --ignore-path bc:/spec/output/imageLabels --ignore-path bc:/spec/strategy/dockerStrategy/buildArgs ${paramFile} --param TAGVERSION=${context.tagversion}"
```

Comment:
This code expects that component to have an app label of the form <projectId>-<componentId>. Typical labels I have seen in quickstarters looked more like this:

```yaml
   labels:
      app: ${PROJECT}
      component: ${COMPONENT}
      env: ${ENV}
  ```

Is this intended here?

Reply:
You are correct, the labels created by quickstarters look like you said. I have to adapt it here, or we need to change it when creating components. I used the labels as I have them here because that is the format we use in our initiative.

Thinking a bit more about this, I thought that we might want to put a `Tailorfile` into the `openshift` folder, and specify the label in there. Then it is easy to change for the user.